### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by nunnatsa**
The primary goal of this PR is to bump the dependencies. In attempting to bump the dependences @nunnatsa discovered a conflict between kubevirt/client-go and updating the core k8s depenedences. Upon further inspection, it was determined that we only needed the kubevirt/client-go package for our e2e tests, and not the capk controllers themselfs.

So, in order to update the core k8s dependences for the capk controllers without entering into dependency hell, Nahshon separated the dependences for the e2e tests and capk controllers. This removed us from having to have kubevirt/client-go in our primary go.mod file.

In all, his changes are...

* Separate dependencies for the controllers and e2e into separate go.mod files
* Update the code to work with the new dependencies. (that meant moving the validation webhook logic and passing around the context object)
* Removing Passt support (the e2e test is failing and we don't intend to support passt binding anymore)

```release-note
Bump dependencies and separate e2e test and controller dependencies
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
> this PR does a bit more than just bumping dependencies. perhaps it's worth splitting it to several focused PRs @nunnatsa, to simplify the review? or, please add more comments about various updates you're making to the logic. thanks

@agradouski - there is no logic change. All the code changes are required by API changes in the bumped modules. I don't know how to split it to smaller commits, if I want any commit to be compile-able.

File: controllers/kubevirtmachine_controller.go [hunk 1]
```diff
@@ -152,9 +151,11 @@ func (r *KubevirtMachineReconciler) Reconcile(goctx gocontext.Context, req ctrl.
 	// Always attempt to Patch the KubevirtMachine object and status after each reconciliation.
 	defer func() {
 		if err := machineContext.PatchKubevirtMachine(patchHelper); err != nil {
-			machineContext.Logger.Error(err, "failed to patch KubevirtMachine")
-			if rerr == nil {
-				rerr = err
+			if err = utilerrors.FilterOut(err, apierrors.IsNotFound); err != nil {
```
Comment (davidvossel) [inline controllers/kubevirtmachine_controller.go:154]: this filter for IsNotFound shouldn't be necessary here.

File: kubevirtci [hunk 2]
```diff
@@ -2,20 +2,20 @@
 
 set -e
 
-export KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER:-k8s-1.23}
-export TENANT_CLUSTER_KUBERNETES_VERSION=${TENANT_CLUSTER_KUBERNETES_VERSION:-v1.24.9}
-export KUBEVIRTCI_TAG=${KUBEVIRTCI_TAG:-2205231118-f12b50e}
+export KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER:-k8s-1.27}
+export TENANT_CLUSTER_KUBERNETES_VERSION=${TENANT_CLUSTER_KUBERNETES_VERSION:-v1.27.6}
```
Comment (davidvossel) [inline kubevirtci:6]: i don't think we have a ubuntu image for v1.27 yet. The highest image i see is `quay.io/capk/ubuntu-2004-container-disk:v1.26.0`

File: pkg/webhookhandler/validator.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	// Return allowed if everything succeeded.
+	return admission.Allowed("")
+}
+
+func (wh *kubevirtMachineTemplateHandler) validateUpdate(old *v1alpha1.KubevirtMachineTemplate, requested *v1alpha1.KubevirtMachineTemplate) error {
+	if !reflect.DeepEqual(old.Spec, requested.Spec) {
+		return errors.New(immutableWarning)
+	}
```
Comment (davidvossel) [inline pkg/webhookhandler/validator.go:100]: We don't need to do this for this PR. But if all we care about is immutability (which appears to be the case) we can enforce this on the CRD now with CEL, and avoid a webhook entirely.

**Comment by davidvossel** [comment]
>> this PR does a bit more than just bumping dependencies. perhaps it's worth splitting it to several focused PRs @nunnatsa, to simplify the review? or, please add more comments about various updates you're making to the logic. thanks
> @agradouski - there is no logic change. All the code changes are required by API changes in the bumped modules. I don't know how to split it to smaller commits, if I want any commit to be compile-able.

Year, there are a few things going on here. Here's some further context.

The primary goal of this PR is to bump the dependencies. In attempting to bump the dependences @nunnatsa discovered a conflict between `kubevirt/client-go` and updating the core k8s depenedences. Upon further inspection, it was determined that we only needed the `kubevirt/client-go` package for our e2e tests, and not the capk controllers themselfs.

So, in order to update the core k8s dependences for the capk controllers without entering into dependency hell, Nahshon separated the dependences for the e2e tests and capk controllers. This removed us from having to have `kubevirt/client-go` in our primary go.mod file.

In all, his changes are...
1. Separate dependencies for the controllers and e2e into separate go.mod files
2. Update the code to work with the new dependencies. (that meant moving the validation webhook logic and passing around the context object)
3. Removing Passt support (the e2e test is failing and we don't intend to support passt binding anymore)

**Review by davidvossel — COMMENTED** [review]
/lgtm

I updated the description of the PR to  better reflect the work that is being done here. This PR is important because it gives us the ability more easily bump dependencies in the future for the controllers while not conflicting with some of the dependencies required for our e2e tests.

**Pull request merged**